src/hooks/useBestAPY.ts

```diff
@@ -0,0 +1,132 @@
+import { Pair, Percent } from '@swapr/sdk'
+import { useMemo } from 'react'
+import { DateTime, Duration } from 'luxon'
+import { useQuery, gql } from '@apollo/client'
+import { getBestApyPairCampaign } from '../utils/liquidityMining'
+import { useActiveWeb3React } from '.'
+import { SubgraphLiquidityMiningCampaign } from '../apollo'
+import { useAllTokensFromActiveListsOnCurrentChain } from '../state/lists/hooks'
+import { useNativeCurrency } from './useNativeCurrency'
+import { useKpiTokens } from './useKpiTokens'
+import { SubgraphToken } from '../utils'
+import { getTokenAmount, getPairWithLiquidityMiningCampaign } from '../utils/index'
+
+const QUERY = gql`
+  query($lowerTimeLimit: BigInt!, $pairAddress: ID) {
+    pair(id: $pairAddress) {
+      address: id
+      reserve0
+      reserve1
+      reserveNativeCurrency
+      totalSupply
+      token0 {
+        address: id
+        name
+        symbol
+        decimals
+      }
+      token1 {
+        address: id
+        name
+        symbol
+        decimals
+      }
+      liquidityMiningCampaigns(where: { endsAt_gt: $lowerTimeLimit }) {
+        address: id
+        duration
+        startsAt
+        endsAt
+        locked
+        stakingCap
+        rewards {
+          token {
+            address: id
+            name
+            symbol
+            decimals
+            derivedNativeCurrency
+          }
+          amount
+        }
+        stakedAmount
+      }
+    }
+  }
+`
+
+interface SubgraphPair {
+  address: string
+  reserve0: string
+  reserve1: string
+  reserveNativeCurrency: string
+  reserveUSD: string
+  totalSupply: string
+  token0: SubgraphToken
+  token1: SubgraphToken
+  liquidityMiningCampaigns: SubgraphLiquidityMiningCampaign[]
+}
+
+interface QueryResult {
```

Comment:
This one is different. One receives an array, the other receives only one `Pair`

Reply:
Can be generic type then:

```ts

```

> > Since reusable chunks that you extracted are both related to liquidityMining how about moving them to `utils/liquidityMining` instead of `utils/index`?
> 
> still not sure if this code fits in `utils` folder.

I agree but can't think of anything better :D